src/Lean/Widget/InteractiveCode.lean

```diff
@@ -82,7 +82,8 @@ def ppExprTagged (e : Expr) (explicit : Bool := false) : MetaM CodeWithInfos :=
       withOptionAtCurrPos pp.explicit.name true do
         delabApp
     else
-      delab
+      withOptionAtCurrPos pp.proofs.name true do
```

Comment:
@Vtec234 Does this change seem fine? It causes just the top-level of delaboration to show the proof term. I made this change since otherwise if you hover over the `⋯` in `(⋯ : 1 ≤ Nat.succ 1)`, you just see something like `(⋯ : 1 ≤ Nat.succ 1) : 1 ≤ Nat.succ 1`.

There might be another design where OmissionInfo is processed differently for top-level expressions.

Reply:
Hm, `ppExprTagged` is used to print a bunch of things. This includes the target type, and hypothesis types/values, in tactic states, as well as expressions in widgets. So changing it has global consequences. But I think it's a defensible choice to set `pp.proofs := true` at the root position, since if the only thing being printed in a given situation is a proof, you probably want to show that proof and not ⋯. In that case I guess you want to set it regardless of `explicit` since `explicit` doesn't seem to imply `proofs`.

Reply:
Thanks for taking a look.

The function that's the main frontend to the delaborator, `delabCore`, does something stronger, which is if the thing being pretty printed is a proof, then [it overrides the option `pp.proofs` and sets it to true](https://github.com/leanprover/lean4/blob/43bbedca46f890e2d2b29d92f71b1e7b76aa0e93/src/Lean/PrettyPrinter/Delaborator/Basic.lean#L343-L347). I had thought about doing that here, but I wasn't sure and left it as a positional option.